test/s3path.jl

```diff
@@ -354,4 +355,39 @@ end
     rm(json_path)
 end
 
+@testset "S3Path versioning" begin
+    s3_enable_versioning(aws, bucket_name)
+    key_version_file = "test_versions"
+    s3_put(aws, bucket_name, key_version_file, "data.v1")
+    s3_put(aws, bucket_name, key_version_file, "data.v2")
+
+    # `s3_list_versions` returns versions in the order newest to oldest
+    versions = [d["VersionId"] for d in reverse!(s3_list_versions(aws, bucket_name, key_version_file))]
+    @test length(versions) == 2
+    @test read(S3Path(bucket_name, key_version_file; config=aws, version=first(versions)), String) == "data.v1"
+    @test read(S3Path(bucket_name, key_version_file; config=aws, version=last(versions)), String) == "data.v2"
+    @test isequal(read(S3Path(bucket_name, key_version_file; config=aws, version=last(versions)), String),
+                  read(S3Path(bucket_name, key_version_file; config=aws), String))
+    @test isequal(read(S3Path(bucket_name, key_version_file; config=aws, version=last(versions)), String),
+                  read(S3Path(bucket_name, key_version_file; config=aws, version=nothing), String))
+
+    unversioned_path = S3Path(bucket_name, key_version_file; config=aws)
+    versioned_path = S3Path(bucket_name, key_version_file; config=aws, version=last(versions))
+    @test versioned_path.version == last(versions)
+    @test unversioned_path.version === nothing
+    @test exists(versioned_path)
+    @test exists(unversioned_path)
+    nonexistent_versioned_path = S3Path(bucket_name, key_version_file; config=aws, version="feVMBvDgNiKSpMS17fKNJK3GV05bl8ir")
+    @test !exists(nonexistent_versioned_path)
+
+    @test S3Path("s3://$(bucket_name)/$(key_version_file)"; version=first(versions)).version == first(versions)
```

Comment:
Something about this makes my eyes bleed
```suggestion
    versioned_path_v1 = S3Path("s3://$(bucket_name)/$(key_version_file)"; version=first(versions)
    @test versioned_path_v1.version == first(versions)
```